### ISSUE TITLE
[IDEA] Allow Comments inside $let

### ISSUE THREAD

**Issue by kookma** [Jermolene/TiddlyWiki5#8212]
**Is your feature request related to a problem? Please describe.**
$let and its ability to create variables from other variables defined  above the mentioned var makes $let an essential and very common widget.

For example I have $let like below:

```xml
<$let 
 dataType =     {{{ [<fields>!is[blank]then[field]] [<indexes>!is[blank]then[index]] +[first[]] }}}
 coulmnFilter = {{{ [<dataType>match[field]then<fields>else<indexes>] }}} 
 columnsCount = {{{ [subfilter<coulmnFilter>count[]] }}}
 
 currentTable =    {{{ [<stateTiddler>!is[blank]else<currentTiddler>]  }}}
 tempTable   =     {{{ [<baseState>][<currentTable>]    +[join[/]] }}}
 tempTableSort =   {{{ [<tempTable>] "sortby"           +[join[/]] }}}
 tempTableExpand = {{{ [<tempTable>] "expand"           +[join[/]] }}}
 tempTableEdit =   {{{ [<tempTable>] "edit-view-status" +[join[/]] }}}
 
 sortIndex =    {{{ [<tempTableSort>getindex[sortIndex]]  }}} 
 sortFlag  =    {{{ [<tempTableSort>getindex[sortFlag]]   }}}

```

The current implementation of $let does not allow to include comments `<!-- coment -->` inside the $let, while for long and large number of variables defined in single $let having comments are very desirable and makes code more clear.

Strangely if I use comments like `<!-- comment  --` without closing tag, it works with current Implementation

**Describe the solution you'd like**
I like TW allows comments at least single line (or inline) comments inside $let widget

**Comment by pmario** [comment]
I think all of this can easier be defined using `\funciton`

```
\function f.dataType()  [<fields>!is[blank]then[field]] [<indexes>!is[blank]then[index]] +[first[]]
\function f.coulmnFilter() [<dataType>match[field]then<fields>else<indexes>]
... and so on 
```

**Comment by Jermolene** [comment]
Thank you @kookma I think we should consider this as part of #8191 and try to develop a congruent syntax for both situations.

```
<$let 
 // This is a comment
 dataType =     {{{ [<fields>!is[blank]then[field]] [<indexes>!is[blank]then[index]] +[first[]] }}}
 coulmnFilter = {{{ [<dataType>match[field]then<fields>else<indexes>] }}} 
 columnsCount = {{{ [subfilter<coulmnFilter>count[]] }}} // This is also a comment
 
 currentTable =    {{{ [<stateTiddler>!is[blank]else<currentTiddler>]  }}}
 tempTable   =     {{{ [<baseState>][<currentTable>]    +[join[/]] }}}
 tempTableSort =   {{{ [<tempTable>] "sortby"           +[join[/]] }}}
 tempTableExpand = {{{ [<tempTable>] "expand"           +[join[/]] }}}
 tempTableEdit =   {{{ [<tempTable>] "edit-view-status" +[join[/]] }}}
 
 sortIndex =    {{{ [<tempTableSort>getindex[sortIndex]]  }}} 
 sortFlag  =    {{{ [<tempTableSort>getindex[sortFlag]]   }}}
>
```

**Comment by pmario** [comment]
>Thank you @kookma I think we should consider this as part of https://github.com/Jermolene/TiddlyWiki5/pull/8191 and try to develop a congruent syntax for both situations.

```
<$let 
 // This is a comment
 dataType =     {{{ [<fields>!is[blank]then[field]] [<indexes>!is[blank]then[index]] +[first[]] }}}
...
```

I think, parsing this comment will be part of the widget parameter parser. So IMO it's a completely different part of the code.
#8191 is part of `fitlers.js` `parseOperand()`

@Jermolene 

IMO The suggested comment would be part of `.parseParameterDefinition()` in `parseutils.js` and should be a completely different PR.

**Comment by Jermolene** [comment]
> I think, parsing this comment will be part of the widget parameter parser. So IMO it's a completely different part of the code. #8191 is part of `fitlers.js` `parseOperand()`

Yes, indeed, that is clearly the case. But these two syntaxes are solving the same problem and so the user experience needs to be consistent.

> IMO The suggested comment would be part of ~`.parseParameterDefinition()`~ `.parseAttribute()` in `parseutils.js` and should be a completely different PR.

For clarity, I am not saying that they should be the same PR, I am just saying that we should fully design both syntax before we proceed with either implementation. I don't want to implement comments in filters and have that locked in before we then discover that we are obliged to take a confusingly subtly different approach to the syntax for comments within widgets.

**Comment by pmario** [comment]
>For clarity, I am not saying that they should be the same PR, I am just saying that we should fully design both syntax before we proceed with either implementation. I don't want to implement comments in filters and have that locked in before we then discover that we are obliged to take a confusingly subtly different approach to the syntax for comments within widgets.

Ahh, OK
